test/jest/batchUploader.spec.ts

```diff
@@ -104,4 +113,43 @@ describe('BatchUploader', () => {
             expect(secondCallTime).toBe(firstCallTime);
         });
     });
+
+    describe('noFunctional', () => {
+        beforeEach(() => {
+            localStorage.clear();
+            sessionStorage.clear();
+        });
+
+        it('should disable offline storage when noFunctional is true', () => {
+            mockMPInstance._Store.noFunctional = true;
+
+            const uploader = new BatchUploader(mockMPInstance, 1000);
+            // @ts-expect-error private
+            expect(uploader.offlineStorageEnabled).toBe(false);
+
+            uploader.queueEvent({ EventDataType: 4 } as any);
+            expect(sessionStorage.getItem('mprtcl-v4_abcdef-events')).toBeNull();
+            expect(localStorage.getItem('mprtcl-v4_abcdef-batches')).toBeNull();
+        });
+
+        it('should enable offline storage when noFunctional is default (false)', () => {
+            // mockMPInstance._Store.noFunctional = false;
+            const uploader = new BatchUploader(mockMPInstance, 1000);
+
+            // @ts-expect-error private
+            expect(uploader.offlineStorageEnabled).toBe(true);
```

Comment:
You may be able to get rid of this override by accessing the private attribute like this:

```suggestion
            expect(uploader['offlineStorageEnabled']).toBe(true);
```